### DISCUSSION TITLE
request to https://chakra-v3-docs.vercel.app/compositions/index.json failed

### DISCUSSION THREAD

**Discussion by ziyiyu** [chakra-ui/chakra-ui#9266] [General] [answered]
When you run the command npx @chakra-ui/cli snippet add, the following error will occur.
file:///C:/workspace/gambo/guangjin-frontend/node_modules/node-fetch/src/index.js:108
                        reject(new FetchError(`request to ${request.url} failed, reason: ${error.message}`, 'system', error));
FetchError: request to https://chakra-v3-docs.vercel.app/compositions/index.json failed, reason: 
    at ClientRequest.<anonymous> (file:///C:/workspace/gambo/guangjin-frontend/node_modules/node-fetch/src/index.js:108:11)
    at ClientRequest.emit (node:events:519:28)
    at emitErrorEvent (node:_http_client:101:11)
    at TLSSocket.socketErrorListener (node:_http_client:504:5)
    at TLSSocket.emit (node:events:519:28)
    at emitErrorNT (node:internal/streams/destroy:169:8)
    at emitErrorCloseNT (node:internal/streams/destroy:128:3)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  type: 'system',
  errno: 'ETIMEDOUT',
  code: 'ETIMEDOUT',
  erroredSysCall: undefined
}
This problem has troubled me for a day until I downloaded the source code and saw the following part.
`const env = processEnvSchema.parse(process.env)

const agent = env.HTTPS_PROXY ? new HttpsProxyAgent(env.HTTPS_PROXY) : undefined

export async function fetchCompositions() {
  const res = await fetch(`${env.REGISTRY_URL}/compositions/index.json`, {
    agent,
  })
  const json = await res.json()
  return compositionIndexSchema.parse(json)
}`
We can solve this problem by setting a proxy in the environment variables.
We can first run the following command.
`$env:HTTPS_PROXY="http://10.35.109.142:8088"`
"http://10.35.109.142:8088" is a proxy address that can access https://chakra-v3-docs.vercel.app/compositions/index.json.

If you are using the Apple system, you can run the following command.
` export HTTPS_PROXY=http://10.35.109.142:8088`
At this point, run 
`npx @chakra-ui/cli snippet add,`
![image](https://github.com/user-attachments/assets/aa007020-ce7c-4893-9cc5-deb7397c618a)
This problem seems to be resolved.However, the following content appeared in the last line of the output.

FetchError: request to https://chakra-v3-docs.vercel.app/compositions/button.json failed, reason: Client network socket disconnected before secure TLS connection was established 

Rerunning the command npx @chakra-ui/cli snippet add seems to resolve this issue.

![image](https://github.com/user-attachments/assets/f82c5f20-cfa6-4e76-b07b-13a25eec1c46)

done！

**Comment by cfping** [comment]
export https_proxy=http://127.0.0.1:7890 http_proxy=http://127.0.0.1:7890 all_proxy=socks5://127.0.0.1:7890
I encountered the same problem. I tried setting the proxy first and rerunning the command npx @chakra-ui/cli snippet add, but the issue wasn't resolved.
`file:///Users/otis/dev/DAO/node_modules/.pnpm/node-fetch@3.3.2/node_modules/node-fetch/src/index.js:108
			reject(new FetchError(`request to ${request.url} failed, reason: ${error.message}`, 'system', error));
			       ^

FetchError: request to https://chakra-v3-docs.vercel.app/compositions/index.json failed, reason: connect ETIMEDOUT 31.13.90.33:443
    at ClientRequest.<anonymous> (file:///Users/otis/dev/DAO/node_modules/.pnpm/node-fetch@3.3.2/node_modules/node-fetch/src/index.js:108:11)
    at ClientRequest.emit (node:events:524:28)
    at emitErrorEvent (node:_http_client:104:11)
    at TLSSocket.socketErrorListener (node:_http_client:518:5)
    at TLSSocket.emit (node:events:524:28)
    at emitErrorNT (node:internal/streams/destroy:170:8)
    at emitErrorCloseNT (node:internal/streams/destroy:129:3)
    at process.processTicksAndRejections (node:internal/process/task_queues:90:21) {
  type: 'system',
  errno: 'ETIMEDOUT',
  code: 'ETIMEDOUT',
  erroredSysCall: 'connect'
}

Node.js v22.13.1`

**Reply by yunup666-cloud** [reply]
HTTP_PROXY=http://127.0.0.1:7890 HTTPS_PROXY=http://127.0.0.1:7890 npx @chakra-ui/cli snippet add

This works for me

**Comment by segunadebayo** [comment, marked as answer]
That Vercel URL is gone. Compositions live at https://chakra-ui.com/docs/compositions
